lib/Component.lua

```diff
@@ -121,6 +121,11 @@ function Component:setState(partialState)
 		error(INVALID_SETSTATE_MESSAGE, 0)
 	end
 
+	-- If the partial state is a function, invoke it to get the actual partial state.
+	if type(partialState) == "function" then
+		partialState = partialState(self.props, self.state)
```

Comment:
Let's flip-flop the arguments to match React. I think most cases of functional `setState` are going to depend on `state` more than `props` anyhow.

Reply:
Makes sense; fixed now!